Subject: [PATCH] Delay move input method windows when exiting.

If we have a dialog with the IME shown, when leaving split-screen mode,
the IME could not find the right window because of early called
WMS.findDesiredInputMethodWindowIndexLocked, bad user experience.

To fix this, delay calling WMS.moveInputMethodWindowsIfNeededLocked().

diff --git a/services/core/java/com/android/server/wm/WindowManagerService.java b/services/core/java/com/android/server/wm/WindowManagerService.java
index 3687512..3660f65 100644
--- a/services/core/java/com/android/server/wm/WindowManagerService.java
+++ b/services/core/java/com/android/server/wm/WindowManagerService.java
@@ -2565,10 +2565,6 @@
 
         win.mRemoved = true;
 
-        if (mInputMethodTarget == win) {
-            moveInputMethodWindowsIfNeededLocked(false);
-        }
-
         if (false) {
             RuntimeException e = new RuntimeException("here");
             e.fillInStackTrace();
@@ -2583,6 +2579,10 @@
         mPolicy.removeWindowLw(win);
         win.removeLocked();
 
+        if (mInputMethodTarget == win) {
+            moveInputMethodWindowsIfNeededLocked(false);
+        }
+
         if (DEBUG_ADD_REMOVE) Slog.v(TAG_WM, "removeWindowInnerLocked: " + win);
         mWindowMap.remove(win.mClient.asBinder());
         if (win.mAppOp != AppOpsManager.OP_NONE) {
